axelor-purchase/src/main/java/com/axelor/apps/purchase/service/PurchaseOrderLineServiceImpl.java

```diff
@@ -252,6 +252,14 @@ public PurchaseOrderLine fill(PurchaseOrderLine line, Product product) throws Ax
     TaxLine taxLine = getTaxLine(order, line);
     line.setTaxLine(taxLine);
 
+    if (!Beans.get(AppPurchaseService.class).getAppPurchase().getManageSupplierCatalog()
+        || product.getSupplierCatalogList().isEmpty()) {
+      line.setProductCode(product.getCode());
+      line.setProductName(product.getName());
+      line.setPrice(product.getPurchasePrice());
+      return line;
+    }
+
     BigDecimal price = getExTaxUnitPrice(order, line, taxLine);
     BigDecimal inTaxPrice = getInTaxUnitPrice(order, line, taxLine);
     String productName = getProductSupplierInfos(order, line)[0];
```

Comment:
Check individual method where check done on supplier catalog and return proper value from there if no catalog  configuration is not enabled.